lib/virt_autotest/utils.pm

```diff
@@ -140,6 +140,8 @@ sub check_failures_in_journal {
         else {
             record_soft_failure("Found new failures: Fake bsc#5555(by PR rule)\n" . $failures . "This is an unknown failure which need to be investigated!");
         }
+
+        upload_system_logs;
```

Comment:
I think calling ```upload_system_logs``` here directly is not very good, because ```check_failures_in_journal``` does not serve the logs uploading purpose by design. And ```upload_system_logs``` does not do as expected when it comes to virtual machine. I think you can use it more wisely and also simply as below:
```
upload_system_logs if (check_failures_in_journal returns some failures);
```
This does not only server your purpose, but also keep everything tidy and clear.  Calling ```upload_system_logs``` inside introduces coupling and, at the same time, make the original function very difficult to use and to be called by other functions, especially when there are repeated calls.

Reply:
Healthy checking let user know when "the error message" firstly occurred, which is the most important. For example, you firstly spot a call trace starts from a certain test module. Then you can postpone logs collecting at the end of the whole test suite by using test module:
```
tests/virt_autotest/validate_system_health.pm
```
This is the original design. ```check_host_health``` and ```check_guest_health``` are in base module which means ```check_failures_in_journal``` is also frequently called at the beginning of each test module chained together. It is not very meaningful for the test to upload the same kind of logs for each module from my POV.

Reply:
No. Always upload logs *when you check them*. The test can fail long before it reaches `validate_system_health` and then you'll have nothing to debug with, especially when the error is hard to reproduce.